### PR TITLE
[CHORE] Remove old stats/logs pages + typescript

### LINKED ISSUE #4475: [CHORE] Remove old stats page and logs
In #4474 we remove calls to a lot of backend code that we used to create stats and query logs. This should be removed because it has been replaced by the new dashboard (and the logs were actually only used ever by me, and I can also query the real DB).

How to get started with this issue:

* Python: The calls we have removed call the routes `/stats/class/` and `/logs/class/`, so you could search the Python code and find references to these. They are in statistics.py, but note that that file also contains routes related to the grid, which *is* still in use
* Typescript: I know there is also code in the front-end to render graphs, but I am not 100% sure where. Figuring that out is step 2 :)

### PR DESCRIPTION
**Pull request by jtwaleson**
**Description**

Removes outdated resources (html, routes and typescript) after #4474 was merged.

Fixes #4475

**How to test**

- CI should cover most
- Check if stats pages for teachers and admins work

**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)

- [x] Contains one of the PR categories in the name
- [x] Describes changes in the format above
- [x] Links to an existing issue or discussion
- [x] Has a "How to test" section

### CODE REVIEW

**Comment by jtwaleson** [comment]
I'm not sure yet about a couple of things:

1. In the admin-classes.html page I just replaced the link to /stats/class/{id} to /live_stats/class/{id} . Not sure if that us 1:1 replaceable.
2. Is there more code that is now obsolete?
3. Did I not remove anything that is still used in a way that I was not able to find.

**Comment by Felienne** [comment]
Set this to draft officially so we know not to merge

File: static_babel_content.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
...
-LANGUAGE_NAMES = {'ca': 'Català', 'da': 'Dansk', 'sw': 'Kiswahili', 'de': 'Deutsch', 'sv': 'Svenska', 'pt_BR': 'Português (Brasil)', 'uk': 'Українська', 'cy': 'Cymraeg', 'zh_Hans': '中文 (简体)', 'fr': 'Français', 'hu': 'Magyar', 'he': 'עברית', 'eo': 'Esperanto', 'es': 'Español', 'tl': 'Filipino (Pilipinas)', 'bg': 'Български', 'it': 'Italiano', 'th': 'ไทย', 'ar': 'العربية', 'sq': 'Shqip', 'ja': '日本語', 'ru': 'Русский', 'fy': 'Frysk',
```
Comment (jpelay) [inline static_babel_content.py:13]: Is this a unintended change?

File: static/js/statistics.ts [hunk 2]
```diff
@@ -564,39 +74,6 @@ export function InitLineChart(data: any[], labels: any[]){
   });
 }
 
-export function enable_level_class_overview(level: string) {
```
Comment (jpelay) [inline static/js/statistics.ts:567]: This function is in use!

File: static/js/statistics.ts [hunk 3]
```diff
@@ -564,39 +74,6 @@ export function InitLineChart(data: any[], labels: any[]){
...
-  if ($('#level_button_' + level).hasClass('gray-btn')) {
-      $('#level_button_' + level).removeClass('gray-btn');
-      $('#level_button_' + level).addClass('green-btn');
-  } else {
-      $('#level_button_' + level).removeClass('green-btn');
-      $('#level_button_' + level).addClass('gray-btn');
-  }
-}
-
-export function select_levels_class_overview(class_id: string) {
```
Comment (jpelay) [inline static/js/statistics.ts:577]: This one too

File: static/js/statistics.ts [hunk 4]
```diff
@@ -608,31 +85,4 @@ export function toggle_show_students_class_overview(adventure: string) {
   }
 }
 
-export function getRunsOverTime(data: any[], labels: any[]) {
```
Comment (jpelay) [inline static/js/statistics.ts:611]: So is this one

Comment (jpelay) [reply static/js/statistics.ts:611]: In general, the lines that says that they were edited by boryana are safe to delete, and the ones that are more recent are still in use

File: templates/admin/admin-classes.html [hunk 5]
```diff
@@ -37,7 +37,7 @@ <h4 class="text-center">Total amount of shown classes: {{ classes|length }} <br>
             <td class="error_percentage_week_cell">{% if class.stats.week.runs > 0 %}{{(class.stats.week.fails / class.stats.week.runs * 100)|round(2)}}{% else %}0.00{% endif %}%</td>
             <td class="runs_cell">{{class.stats.total.runs}}</td>
             <td class="error_percentage_cell">{% if class.stats.total.runs > 0 %}{{(class.stats.total.fails / class.stats.total.runs * 100)|round(2)}}{% else %}0.00{% endif %}%</td>
-            <td id="statistics_cell"><a href="/stats/class/{{class.id}}" target="_blank">Click here</a></td>
+            <td id="statistics_cell"><a href="/live_stats/class/{{class.id}}" target="_blank">Click here</a></td>
```
Comment (jpelay) [inline templates/admin/admin-classes.html:40]: This is an error that was there before you edited this files, because I just tested it in main and it also failed, but this button made it apparent. When trying to access the class is returning this error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2213, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2193, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/workspace/website/auth.py", line 213, in inner
    return f(*args, user=current_user(), **kws)
  File "/workspace/website/statistics.py", line 343, in render_live_stats
    adventures = self.get_class_live_stats(user, class_)
  File "/workspace/website/statistics.py", line 378, in get_class_live_stats
    adventures = self.__get_adventures_for_overview(user, class_['id'])
  File "/workspace/website/statistics.py", line 327, in __get_adventures_for_overview
    return _get_available_adventures(adventures, teacher_adventures, customizations, last_adventures)
  File "/workspace/website/statistics.py", line 1119, in _get_available_adventures
    adventure_name = adventure_names[adventure_key]
KeyError: '3f8aea42eb324f08a16776671498dd1b'
```

It doesn't happen when the user is `teacher1` so this is probably an error in the way we are retrieving the teacher adventures from the database. Would you like to tackle this in this PR? If not, there's no problem! I can also open a separate issue for this and work on it later.

**Comment by jpelay** [comment]
Hi @jtwaleson! Thank you for this PR!! It's great and it does help a lot in improving our codebase! 

> I'm not sure yet about a couple of things:
> 
> 1. In the admin-classes.html page I just replaced the link to /stats/class/{id} to /live_stats/class/{id} . Not sure if that us 1:1 replaceable.

I think that in order to replace `/stats/` for `/live_stats` we need to change a little bit the way we retrieve the items from the DB, because since the user is `admin` and not the teacher who created the class, it generates the error I described in an earlier comment. At least that's my hunch! 

> 2. Is there more code that is now obsolete?

I think that you removed all of it!

> 3. Did I not remove anything that is still used in a way that I was not able to find.

While reviewing I thought you did, but after further inspection you actually removed some code that I missed to remove when changing some things here! Thank you! And sorry for the confusion!!

File: static_babel_content.py (hunk 1 above)
Comment (jtwaleson) [reply static_babel_content.py:13]: Hm yes, this just happened after I ran `npm run build` locally.

Comment (Felienne) [reply static_babel_content.py:13]: Yes we generate a few files locally (like app.js also) but you don't have to commit it in a PR. This process is also ran in an action, if I recall correctly (feel free to double check this, and to add it to the docs if you think that is useful!).

File: templates/admin/admin-classes.html (hunk 5 above)
Comment (Felienne) [reply templates/admin/admin-classes.html:40]: If we are cleaning up, we might as well also delete all code related to the classes button for the admin: 
image

(in this PR or in another one)

Since there are soooo many classes in the database, this page often crashes so I never use it and look at my local copy of the database if I want to find info.

**Comment by Felienne** [comment]
Thanks for the progress @jtwaleson! I think the test that are now failing (the admin can also see stats) can now be removed and then maybe we are good to go?

**Comment by jtwaleson** [comment]
> Thanks for the progress @jtwaleson! I think the test that are now failing (the admin can also see stats) can now be removed and then maybe we are good to go?

Still work in progress! I'll let you know when it's ready for final review.

File: templates/admin/admin-classes.html (hunk 5 above)
Comment (jtwaleson) [reply templates/admin/admin-classes.html]: Great, I fixed both the bug and removed the classes overview.

**Comment by Felienne** [comment]
> ~@Felienne It's ready!~
> 
> tests are failing..

Ahhhh, that happens! Let's see where we are at with the tests now!

**Comment by jtwaleson** [comment]
Done.

File: .github/workflows/cypresstests.yml [hunk 6]
```diff
@@ -19,7 +19,7 @@ jobs:
       with:
         fetch-depth: 0
     - name: Set up Python 3.9
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
```
Comment (Felienne) [inline .github/workflows/cypresstests.yml:22]: Hi @jtwaleson! I am not sure if this is an intentional change. Do you use new features in this PR that only exist in v4? If not, can we do this in a separate PR for traceability?

**Comment by jtwaleson** [comment]
I noticed a warning for a node js version installation, but tbh didn’t
check if it was resolved with v4. I’ll rebase it into a separate PR!

File: .github/workflows/cypresstests.yml (hunk 6 above)
Comment (jtwaleson) [reply .github/workflows/cypresstests.yml]: Rebased and will create a separate PR with some GH Actions cleanup at some point.

**Review by Felienne — APPROVED** [review]
Thanks a bunch for the large cleanup!!
